ORT fails to run if git remote URL is not set

## Describe the bug

Hey! I was just trying ORT out, and it seems that it fails on a fresh repository where `git remote` URL is not yet set up. While it's unlikely scenario that this will be empty in real-world use-case, it took me a minute to realize it's not the project that's wrong, but a repo's git config:

```
Exception in thread "main" java.io.IOException: Unable to get the remote URL.
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.getRemoteUrl$lambda$0(GitWorkingTree.kt:119)
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.useRepo(GitWorkingTree.kt:58)
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.getRemoteUrl(GitWorkingTree.kt:99)
	at org.ossreviewtoolkit.downloader.WorkingTree.getInfo(WorkingTree.kt:40)
	at org.ossreviewtoolkit.downloader.VersionControlSystem$Companion.getPathInfo(VersionControlSystem.kt:143)
	at org.ossreviewtoolkit.analyzer.PackageManager.resolveDependencies(PackageManager.kt:343)
	at org.ossreviewtoolkit.analyzer.PackageManagerRunner$run$3.invokeSuspend(Analyzer.kt:354)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
Caused by: java.lang.NullPointerException
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.getRemoteUrl$lambda$0(GitWorkingTree.kt:114)
	... 14 more
```

## To Reproduce

Steps to reproduce the behavior:

1. Create new repository, I've used `pnpm create vite`.
2. Install dependencies, if needed.
3. Run ORT analyze
```
docker run --rm -t \
  -v "$PWD":/work \
  -v ./config.yml:/home/ort/.ort/config/config.yml \
  -w /work \
  ghcr.io/oss-review-toolkit/ort:latest \
  --info analyze \
    -i /work \
    -o /work/ort-out/analyzer \
    -f YAML
```
4. See the error.

## Expected behavior

Exception is handled, with a more descriptive error that the Git remote ULR is missing.

## Console / log output

Full log:

```
❯ ./generate-sbom.sh
12:48:20.855 [main] INFO  org.ossreviewtoolkit.model.config.OrtConfiguration - Using ORT configuration file '/home/ort/.ort/config/config.yml'.
Hoplite is configured to infer which sealed type to choose by inspecting the config values at runtime. This behaviour is now deprecated in favour of explicitly specifying the type through a discriminator field. In 3.0 this new behavior will become the default. To enable this behavior now (and disable this warning), invoke withExplicitSealedTypes() on the ConfigLoaderBuilder.
12:48:21.000 [main] INFO  org.ossreviewtoolkit.utils.common.EnvironmentVariableFilter - EnvironmentVariableFilter initialized with denySubstrings = [key, pass, pwd, token, user] and allowNames = [CARGO_HTTP_USER_AGENT, COMPOSER_ALLOW_SUPERUSER, CONAN_LOGIN_ENCRYPTION_KEY, CONAN_LOGIN_USERNAME, CONAN_PASSWORD, CONAN_USERNAME, CONAN_USER_HOME, CONAN_USER_HOME_SHORT, DOTNET_CLI_CONTEXT_ANSI_PASS_THRU, GIT_ASKPASS, GIT_HTTP_USER_AGENT, GRADLE_USER_HOME, HACKAGE_USERNAME, HACKAGE_PASSWORD, HACKAGE_KEY, PWD, USER, USERPROFILE].
 ______________________________                                                         
/        \_______   \__    ___/ The OSS Review Toolkit, version 77.0.0,                 
|    |   | |       _/ |    |    built with JDK 21.0.9+10-LTS, running under Java 21.0.9.
|    |   | |    |   \ |    |    Executing 'analyze' as 'ort' on Linux                   
\________/ |____|___/ |____|    with 12 CPUs and a maximum of 7948 MiB of memory.       
                                                                                        
Environment variables:                                                                 
HOME = /home/ort                                                                       
TERM = xterm                                                                           
JAVA_HOME = /opt/java/openjdk                                                          
ANDROID_HOME = /opt/android-sdk                                                        
                                                                                       
Looking for ORT configuration in the following file:
        /home/ort/.ort/config/config.yml

Looking for analyzer-specific configuration in the following files and directories:
        /work/.ort.yml (does not exist)
        /home/ort/.ort/config/resolutions.yml (does not exist)
The following 27 package manager(s) are enabled:
        Bazel, Bower, Bundler, Cargo, Carthage, CocoaPods, Composer, Conan, Gleam, GoMod, Gradle Inspector, Maven, NPM, NuGet, PIP, Pipenv, PNPM, Poetry, Pub, SBT, SpdxDocumentFile, Stack, Swift Package Manager, Tycho, Unmanaged, Yarn, Yarn 2+
The following 3 package curation provider(s) are enabled:
        DefaultDir, DefaultFile, Spring
Analyzing project path:
        /work
12:48:21.748 [main] INFO  org.ossreviewtoolkit.utils.ort.OrtAuthenticator - Authenticator was successfully installed.
12:48:21.751 [main] INFO  org.ossreviewtoolkit.utils.ort.OrtProxySelector - Proxy selector was successfully installed.
12:48:22.255 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/work/.stash16697943448160000263/node_modules' as it is hard-coded to be ignored.
12:48:22.255 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/work/node_modules' as it is hard-coded to be ignored.
12:48:22.255 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/work/.stash9859993690006837542/node_modules' as it is hard-coded to be ignored.
12:48:22.255 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/work/.git' as it is hard-coded to be ignored.
12:48:22.255 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/work/.stash9678515291988435774/node_modules' as it is hard-coded to be ignored.
12:48:22.255 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/work/.stash15749465029637879401/node_modules' as it is hard-coded to be ignored.
12:48:22.255 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Not analyzing directory '/work/.stash2423550733308403266/node_modules' as it is hard-coded to be ignored.
12:48:22.257 [main] INFO  org.ossreviewtoolkit.plugins.packagemanagers.node.NodePackageManagerDetection - Detected '/work/package.json' to be the root of a(n) PNPM project.
Found 1 PNPM definition file(s) at:
        package.json
Found in total 1 definition file(s) from the following 1 package manager(s):
        PNPM
12:48:22.262 [main] INFO  org.ossreviewtoolkit.analyzer.Analyzer - Calling before resolution hooks for 1 manager(s).
12:48:22.263 [main] INFO  org.ossreviewtoolkit.plugins.packagemanagers.node.NodePackageManager - Project-specific 'node_modules' directory present at '/work/node_modules'.
12:48:22.265 [main] INFO  org.ossreviewtoolkit.utils.common.DirectoryStash - Temporarily moving directory from '/work/node_modules' to '/work/.stash11504959371922157849/node_modules'.
12:48:22.271 [main] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'node --version' in '/work'...
12:48:22.279 [main] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'pnpm --version' in '/work'...
12:48:22.488 [main] INFO  org.ossreviewtoolkit.analyzer.PackageManagerRunner - Starting PNPM analysis.
12:48:22.491 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.PackageManager - Using PNPM to resolve dependencies for path 'package.json'...
12:48:22.495 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'pnpm install --ignore-pnpmfile --ignore-scripts --frozen-lockfile' in '/work'...
12:48:23.225 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'pnpm list --json --only-projects --recursive' in '/work'...
12:48:23.445 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'pnpm list --json --recursive --depth Infinity --prod' in '/work'...
12:48:23.674 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running 'pnpm list --json --recursive --depth Infinity --dev' in '/work'...
Exception in thread "main" java.io.IOException: Unable to get the remote URL.
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.getRemoteUrl$lambda$0(GitWorkingTree.kt:119)
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.useRepo(GitWorkingTree.kt:58)
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.getRemoteUrl(GitWorkingTree.kt:99)
	at org.ossreviewtoolkit.downloader.WorkingTree.getInfo(WorkingTree.kt:40)
	at org.ossreviewtoolkit.downloader.VersionControlSystem$Companion.getPathInfo(VersionControlSystem.kt:143)
	at org.ossreviewtoolkit.analyzer.PackageManager.resolveDependencies(PackageManager.kt:343)
	at org.ossreviewtoolkit.analyzer.PackageManagerRunner$run$3.invokeSuspend(Analyzer.kt:354)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
Caused by: java.lang.NullPointerException
	at org.ossreviewtoolkit.plugins.versioncontrolsystems.git.GitWorkingTree.getRemoteUrl$lambda$0(GitWorkingTree.kt:114)
	... 14 more
```

## Environment

* ORT version: 77.0.0 (Docker)
* Java version: Provided by Docker image
* OS: Linux Ubuntu 24.04 LTS (docker host)

--- a/downloader/src/main/kotlin/WorkingTree.kt
+++ b/downloader/src/main/kotlin/WorkingTree.kt
@@ -37,7 +37,13 @@ abstract class WorkingTree(val workingDir: File, val vcsType: VcsType) {
      * [workingDir] to [getRootPath]. It is not related to the path argument that was used for downloading, and at the
      * example of Git, it does not reflect the (single) path that was cloned in a sparse checkout.
      */
-    open fun getInfo() = VcsInfo(vcsType, getRemoteUrl(), getRevision(), path = getPathToRoot(workingDir))
+    open fun getInfo() =
+        VcsInfo(
+            type = vcsType,
+            url = runCatching { getRemoteUrl() }.getOrDefault(""),
+            revision = getRevision(),
+            path = getPathToRoot(workingDir)
+        )
 
     /**
      * Return the map of nested repositories, for example Git submodules or Git-Repo modules. The key is the path to the

